design/ceph/admission-controller.md

```diff
@@ -0,0 +1,219 @@
+
+The admission controllers intercept requests based on the values given in the configuration.In this config, We have to provide the details on 
+* What resources should it be looking for ? (Pods, Service)
+* What api version and group does it belong to ? *Example : ApiVersion = (v1, v1beta) Group version = (rook.ceph.io, admissionregistration.k8s.io)*
+* What kind of operation should it intercept ? (Create, Update, Delete)
+* A valid base64 encoded CA bundle.
+* What path do we want to send with HTTPs request (/validate, /mutate)
+
+A server should be in place (*with valid TLS certificates*) to intercept any HTTPs request that comes with the above path value. Once intercepted by a server, an AdmissionRequest(https://github.com/kubernetes/api/blob/master/admission/v1beta1/types.go#L40) object is sent through with the resource specifications. 
+
+Here is the part where the server will perform all our validations on the provided resources values and send back an AdmissionResponse with the field Allowed set accordingly to indicate whether it is accepted or rejected.
+
+Here is structure of our response object. 
+
+**AdmissionResponse Object**
+```
+type AdmissionResponse struct {
+	// UID is an identifier for the individual request/response.
+	// This should be copied over from the corresponding AdmissionRequest.
+	UID types.UID `json:"uid" protobuf:"bytes,1,opt,name=uid"`
+
+	// Allowed indicates whether or not the admission request was permitted.
+	Allowed bool `json:"allowed" protobuf:"varint,2,opt,name=allowed"`
+
+	// Result contains extra details into why an admission request was denied.
+	// This field IS NOT consulted in any way if "Allowed" is "true".
+	// +optional
+	Result *metav1.Status `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+
+	// The patch body. Currently we only support "JSONPatch" which implements RFC 6902.
+	// +optional
+	Patch []byte `json:"patch,omitempty" protobuf:"bytes,4,opt,name=patch"`
+
+	// The type of Patch. Currently we only allow "JSONPatch".
+	// +optional
+	PatchType *PatchType `json:"patchType,omitempty" protobuf:"bytes,5,opt,name=patchType"`
+
+	// AuditAnnotations is an unstructured key value map set by remote admission controller (e.g. error=image-blacklisted).
+	// MutatingAdmissionWebhook and ValidatingAdmissionWebhook admission controller will prefix the keys with
+	// admission webhook name (e.g. imagepolicy.example.com/error=image-blacklisted). AuditAnnotations will be provided by
+	// the admission webhook to add additional context to the audit log for this request.
+	// +optional
+	AuditAnnotations map[string]string `json:"auditAnnotations,omitempty" protobuf:"bytes,6,opt,name=auditAnnotations"`
+}
+```
+
+If any of the controllers in either phase reject the request, the entire request is rejected immediately and an error is returned to the end-user.
+
+## Setup in Rook
+
+### Certificate Management
```

Comment:
Since cert management is not managed by Rook, how about not putting this under the "Setup in Rook" section?

Reply:
Fixed. Added "Starting the webhook server" part under "Setup in Rook"